mod/groups/start.php

```diff
@@ -496,34 +496,45 @@ function groups_create_event_listener($event, $object_type, $object) {
  * Return the write access for the current group if the user has write access to it.
  */
 function groups_write_acl_plugin_hook($hook, $entity_type, $returnvalue, $params) {
-	$page_owner = elgg_get_page_owner_entity();
-	$user_guid = $params['user_id'];
+	
+	if (empty($params) || !is_array($params)) {
```

Comment:
Only use case I can think of is checking if ```$params['user_id'] === 0``` which means that for some reason the hook was called for an non-logged-in user.

Not passing any params seems like something that the developer should be notified about.